x-pack/plugins/lens/public/editor_frame_service/editor_frame/workspace_panel/workspace_panel.tsx

```diff
@@ -148,6 +149,30 @@ export function InnerWorkspacePanel({
     ]
   );
 
+  const variables = useMemo(() => {
+    const indexPatternRefs: Record<string, string> = {};
+
+    Object.entries(datasourceMap).forEach(([datasourceId, datasource]) => {
+      const state = datasourceStates[datasourceId].state;
+      datasource
+        .getMetaData(state)
+        .indexPatternsByLayer.forEach(({ layerId, indexPatternId, refName }, index) => {
+          indexPatternRefs[`filterable-index-pattern-${index}`] = indexPatternId;
+          indexPatternRefs[refName] = indexPatternId;
+        });
+    });
+    return indexPatternRefs;
+  }, [datasourceMap, datasourceStates]);
```

Comment:
@flash1293 this is the key change I've made to your references work. The expression will always be identical in the Lens editor and the embeddable, but it's now up to the expression renderer to provide variables which map all the references to their underlying IDs.